src/net/db.rs

```diff
@@ -456,21 +511,67 @@ impl PeerDB {
         Ok(())
     }
 
-    fn update_local_peer(
+    fn get_schema_version(conn: &Connection) -> Result<String, db_error> {
+        let version = conn
+            .query_row(
+                "SELECT MAX(version) from db_config",
+                rusqlite::NO_PARAMS,
+                |row| row.get(0),
+            )
+            .optional()?
+            .unwrap_or("1".to_string());
+        Ok(version)
+    }
+
+    fn apply_schema_2<'a>(tx: &Transaction<'a>) -> Result<(), db_error> {
+        debug!("Apply schema 2 to peer DB");
+        for row_text in PEERDB_SCHEMA_2 {
+            tx.execute_batch(row_text).map_err(db_error::SqliteError)?;
+        }
+        Ok(())
+    }
+
+    fn apply_schema_migrations<'a>(tx: &Transaction<'a>) -> Result<String, db_error> {
+        debug!("Apply any schema migrations");
+        let expected_version = PEERDB_VERSION.to_string();
+        let mut ret = None;
+        loop {
+            match PeerDB::get_schema_version(tx) {
+                Ok(version) => {
+                    if ret.is_none() {
+                        ret = Some(version.clone());
+                    }
+                    if version == "1" {
+                        PeerDB::apply_schema_2(tx)?;
+                    } else if version == expected_version {
+                        return Ok(ret.expect("unreachable"));
+                    } else {
+                        panic!("The schema version of the peer DB is invalid.")
+                    }
+                }
+                Err(e) => panic!("Error obtaining the version of the peer DB: {:?}", e),
+            }
+        }
```

Comment:
I'd recommend eliminating nesting, the necessity of a mutable option for a non-optional return, and the use of `loop`:

```suggestion
        let begin_ver = PeerDB::get_schema_version(tx).expect("Error obtaining the version of the peer DB");
        let mut version = begin_ver.clone();
        while version != expected_version {
             version = PeerDB::get_schema_version(tx).expect("Error obtaining the version of the peer DB");
             if version == "1" {
                 PeerDB::apply_schema_2(tx)?;
             } else {
                 panic!("The schema version of the Peer DB is invalid. Found = {}", version);
             }
        }
       begin_ver
```

Reply:
Addressed